src/main/scala/org/broadinstitute/hail/expr/Type.scala

```diff
@@ -104,6 +100,12 @@ abstract class Type extends BaseType {
       a => Option(a)
   }
 
+  def printPretty(compact: Boolean = false): String = {
```

Comment:
print sounds like it prints.  Just overload pretty without a StringBuidler.  This should take a prettyAttrs flag, too.  Look at where `pretty` is called in the code and use this where you can.